P0009/P0009.md

```diff
@@ -2255,59 +2239,51 @@ constexpr mapping(const extents_type& e, const span<SizeType, rank>& s) noexcept
+     the integers in the range  $[0,$ _`rank_`_$)$,
      such that `s[` $p_i$ `] >= s[` $p_{i-1}$ `] * e.extent(` $p_{i-1}$ `)` is `true`
-     for all $i$ in the range $[1,$ `rank` $)$. 
+     for all $i$ in the range $[1,$ _`rank_`_ $)$, where $p_i$ is the $i^{th}$ element of $P$. 
      <i>[Note:</i> This condition guarantees that `is_unique()` is `true`. <i>— end note]</i>
 
 [4]{.pnum} *Effects:* Direct-non-list-initializes $extents\_$ with `e`, and for all $d$ in the range 
-                      $[0,$ `rank`$)$, direct-non-list-initializes _`strides_`_`[d]` with `s[d]`.
-
-```c++
-template<class SizeType>
-constexpr mapping(const extents_type& e, const array<SizeType, rank>& s) noexcept;
-```
-[6]{.pnum} *Constraints:*
-
-   * [6.1]{.pnum} `is_convertible_v<const SizeType&, size_type>` is `true`,
-
-   * [6.2]{.pnum} `is_nothrow_constructible_v<size_type, const SizeType&>` is `true`.
-
-[7]{.pnum} *Effects:* Equivalent to: `mapping(e, span{s});`.
+                      $[0,$ _`rank_`_$)$, direct-non-list-initializes _`strides_`_`[`$d$`]` with `as_const(s[`$d$`])`.
 
 ```c++
 template<class StridedLayoutMapping>
   explicit(@_see below_@)
   constexpr mapping(const StridedLayoutMapping& other) noexcept;
 ```
 
-[8]{.pnum} *Constraints:*
+[5]{.pnum} *Constraints:*
 
-   * [8.1]{.pnum} `is_constructible_v<extents_type, StridedLayoutMapping::extents_type>` is `true`.
+   * [5.1]{.pnum} `is_constructible_v<extents_type, typename StridedLayoutMapping::extents_type>` is `true`.
 
-   * [8.2]{.pnum} `StridedLayoutMapping::is_always_unique()` is `true`.
+   * [5.2]{.pnum} `StridedLayoutMapping::is_always_unique()` is `true`.
 
-   * [8.3]{.pnum} `StridedLayoutMapping::is_always_strided()` is `true`.
+   * [5.3]{.pnum} `StridedLayoutMapping::is_always_strided()` is `true`.
 
-   * [8.4]{.pnum} `is_same_v<typename StridedLayoutMapping::extents_type, decltype(other.extents())>` is `true`.
+   * [5.4]{.pnum} `is_same_v<typename StridedLayoutMapping::extents_type, decltype(other.extents())>` is `true`.
 
-   * [8.5]{.pnum} if `rank > 0` then `is_same_v<decltype(other.stride(0)), typename StridedLayoutMapping::size_type>` is `true`.
+   * [5.5]{.pnum} if _`rank_`_` > 0` then `is_same_v<decltype(other.stride(0)), typename StridedLayoutMapping::size_type>` is `true`.
 
-[9]{.pnum} *Precondition:* 
+[6]{.pnum} *Precondition:* 
 
-   * [9.1]{.pnum} `StridedLayoutMapping` meets the layout mapping requirements, 
+   * [6.1]{.pnum} `StridedLayoutMapping` meets the layout mapping requirements, 
 
-   * [9.2]{.pnum} `other.required_span_size()` is a representable value of type `size_type` ([basic.fundamental]), and
+   * [6.2]{.pnum} `other.required_span_size()` is a representable value of type `size_type` ([basic.fundamental]), and
 
-   * [9.3]{.pnum} _`OFFSET`_`(other) == 0` is `true`.
+   * [6.3]{.pnum} _`OFFSET`_`(other) == 0` is `true`.
 
-[10]{.pnum} *Postcondition:* `*this == other` is `true`.
+<!-- TODO static cast other.stride(d) -->
```

Comment:
Will we leave this TODO for another, later PR?

Reply:
this requires the other paper.